wasmplugin/plugin.go

```diff
@@ -110,6 +113,53 @@ func (ctx *httpContext) OnHttpRequestHeaders(numHeaders int, endOfStream bool) t
 	// and its request properties, but they may not be true of other proxies implementing
 	// proxy-wasm.
 
+	if tx.IsRuleEngineOff() {
+		return types.ActionContinue
+	}
+	// default value instantiated, OnHttpRequestHeaders does not terminate if IP/Port retrieve goes wrong
+	var srcAddress, dstAddress AddressInfo
```

Comment:
I would move from line 120 to line 160 into an own function https://github.com/corazawaf/coraza-proxy-wasm/pull/101/files#diff-a755f6a359f42d229feba810f3cd6f9c128aa43ffb716b33b186ec5fe8aa0a49R120-R160 as all the data retrieving seems distracting from the main logic.

Reply:
Thanks, done! I thought about it for a while but had no strong feelings. Adding a parameter, the code is also a bit less redundant. PTAL